public/myScript/orders.js

```diff
@@ -0,0 +1,158 @@
+            case 5:item = 'Отправлен водителю';break;
+            case 6:item = 'Принят водителем';break;
+            case 3:item = 'Отклонен 1 водителем';break;
+            case 2:item = 'Нет ответа от 1 водителя';break;
+            case 1:item = 'Отклонен 2 водителем';break;
+            case 0:item = 'Нет ответа от 2 водителя';break;
+            case 7:item = 'Водитель на месте';break;
+            case 8:item = 'Клиент в машине';break;
+            case 9:item = 'Выполнен';break;
+            case 10:item = 'Отменен';break;
+        }
+        return item;
+    }
+
+    function _buildRowColumns(names, item) {
+        var rowSt = '<tr>';
+        var row ='';
+        if (names && names.length > 0)
+        {
+            $.each(names, function(index, name) {
+                var c = item ? item[name+''] : name;
+                if (name == "status"){
+                    rowSt = format(c);
+                    c=translate(c);
+                }
+                if (c == null) c='';
+                row += '<td>' + c + '</td>';
+            });
+        }
+        rowSt = rowSt +row + '<tr>';
+        return rowSt;
+    }
+
+    /** Builds and sets the headers of the table. */
+    function _setHeaders() {
+        // if no headers specified, we will use the fields as headers.
+        _headers = (_headers == null || _headers.length < 1) ? _fields : _headers;
+        var h = _buildRowColumns(_headers);
+        if (_table.children('thead').length < 1)
+            _table.prepend('<thead bgcolor= "#DCDCDC"></thead>');
+        _table.children('thead').html(h);
+    }
+
+    function _setNoItemsInfo() {
+        if (_table.length < 1) return; //not configured.
+        var colspan = _headers != null && _headers.length > 0 ?
+            'colspan="' + _headers.length + '"' : '';
+        var content = '<tr class="no-items"><td ' + colspan + ' style="text-align:center">' +
+            _defaultText + '</td></tr>';
+        if (_table.children('tbody').length > 0)
+            _table.children('tbody').html(content);
+        else _table.append('<tbody>' + content + '</tbody>');
+    }
+
+    function _removeNoItemsInfo() {
+        var c = _table.children('tbody').children('tr');
+        if (c.length == 1 && c.hasClass('no-items')) _table.children('tbody').empty();
+    }
+
+    return {
+        /** Configres the dynamic table. */
+        config: function(tableId, fields, headers, defaultText) {
+            _tableId = tableId;
+            _table = $('#' + tableId);
+            _fields = fields || null;
+            _headers = headers || null;
+            _defaultText = defaultText || 'No items to list...';
+            _setHeaders();
+            _setNoItemsInfo();
+            return this;
+        },
+        /** Loads the specified data to the table body. */
+        load: function(data, append) {
+            if (_table.length < 1) return; //not configured.
+            _setHeaders();
+            _removeNoItemsInfo();
```

Comment:
According to this, if don't have data in the table, first you will remove "No items" section and then you will add it back again. Refactor this code in such, so it will not make unnecessary changes.